### PR TITLE
Create identity to registry mapping in builder object

### PR DESCRIPTION
**Pull request by pauld-msft**
API implementation of https://msazure.visualstudio.com/DefaultCollection/One/_git/AAPT-Antares-Containerapps/pullRequest/8031995#1684188163

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json [hunk 1]
```diff
@@ -356,6 +356,14 @@
...
           "description": "Resource ID of the compute that the patching will be performed on.",
           "readOnly": true
+        },
+        "identityMappings": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IdentityMappings"
+          },
+          "x-ms-identifiers": [],
+          "description": "List of mappings of managed identities to the container registry that they allow access to."
```
Comment (daniv-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json:366]: Hope I didn't forget about something. So at this point the relationship is 1-1 (one managed identity for one container registry), right?
Please consider updating the description to something like `List of mappings of a managed identity with the container registry that it allows access to. (1-1 relationship)`

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json [hunk 2]
```diff
@@ -439,6 +447,24 @@
...
+        "identityResourceId": {
+          "type": "string",
+          "description": "Resource ID of the managed identity."
+        },
+        "containerRegistryServer": {
+          "type": "string",
+          "description": "Login server of the container registry."
+        }
+      },
+      "description": "Model representing an environment variable.",
```
Comment (daniv-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json:462]: Could you please confirm that this description is up to date? If yes, please consider adding a bit more details (unless this kind of description is normal, in which case feel free to ignore)

**Review by daniv-msft — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json (hunk 2 above)
Comment (pauld-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json:462]: missed updating this. thanks

File: specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json (hunk 1 above)
Comment (pauld-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Builders.json]: Hmm, you do raise a good point of confusion. So we allow for each container registry to have a single managed identity, however, the same managed identity could be used for multiple container registries. So actually, the best data structure for this relationship is going to be a dictionary of container registry -> identity mappings, with the registry login server being the key. I've updated to reflect that new structure

**Comment by pauld-msft** [comment]
merging into other PR in an effort to run the linting / pr jobs

**Pull request merged**